api/v1/tlspolicy_types.go

```diff
@@ -191,7 +191,7 @@ func (p *TLSPolicy) GetRulesHostnames() []string {
 	return make([]string, 0)
 }
 
-// DEPRECATED: Use GetTargetRefs instead
+// Deprecated: kuadrant.Policy.
```

Comment:
"Deprecated:" seems to be the correct way to this in Go https://go.dev/wiki/Deprecated, linter will fail if this is used anywhere.

We can likely remove all these soon anyway, if not already.